lib/record_store/zone/config/ignore_pattern.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RecordStore
+  class Zone
+    class Config
+      class IgnorePattern
+        MATCH_TYPE_FIELD = 'match'
+        MATCH_TYPE_REGEX = 'regex'
+        MATCH_TYPE_EXACT = 'exact'
+
+        def initialize(orig_hash)
+          @orig_hash = orig_hash
+        end
+
+        def ignore?(record)
+          all_pairs.all? do |(key, value)|
+            record.respond_to?(key) && value_matches?(value, record.send(key))
+          end
+        end
+
+        def to_hash
+          @orig_hash
+        end
+
+        private
+
+        def all_pairs
+          to_hash.reject { |key| key == MATCH_TYPE_FIELD }
+        end
+
+        def value_matches?(ignore_pattern_value, record_value)
+          return record_value.match?(ignore_pattern_value) if regex?
```

Comment:
The conditions feel like an OOP antipattern. You were creating two subclasses originally right?